projects/python-client/src/datapane/client/api/user.py

```diff
@@ -106,13 +112,87 @@ def ping(config: t.Optional[c.Config] = None, cli_login: bool = False, verbose:
     return email
 
 
-def _run_script(script: str):
+def _run_script(script: Path):
     """Run the template script and copy it locally to cwd"""
     script_path = ir.files("datapane.resources.templates.hello") / script
     shutil.copyfile(str(script_path), script_path.name)
     runpy.run_path(str(script_path), run_name="__datapane__")
 
 
+def _run_template(template_path: Path):
+    """Run the template script"""
+    # Use template_path as cwd to support relative paths in template script
+    with pushd(template_path):
+        # Try to run the template script.
+        try:
+            runpy.run_path(
+                "template.py",
+                run_name="__datapane__",
+            )
+        # Notify the user of missing packages that are required by the template
+        except ModuleNotFoundError as e:
+            raise DPError(f"Please install the following packages to run this template\n{e.name}") from e
+
+
+def _download_template(url: URL):
+    """Download the template from a repository url, and delete the .git directory
+
+    Returns:
+        The path of the cloned template repository
+    """
+
+    url = _check_repo_url(url)
+
+    # Check if target directory already exists.
+    # This check is identical to the target check in `dulwich.porcelain.clone`:
+    # https://github.com/jelmer/dulwich/blob/78e6ae0960d79060d4ff19f0aa5dc4b32296116d/dulwich/porcelain.py#L439-L441
+    target = url.split("/")[-1]
```

Comment:
Resolves trailing slash problems on the url (`target` will be a blank string).
Happens for some Github URLs and bash autocompletion for local paths.

```suggestion
    target = url.rstrip('/').split("/")[-1]
```